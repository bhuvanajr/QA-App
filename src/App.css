@import 'bootstrap/dist/css/bootstrap.css';
@import './styles/main.css';

:root {
  --qa-green: #354328;
}

#root {
  position: relative;
  padding-bottom: 6.25rem;
  min-height: 100%;
  font-family: Geomanist, Helvetica, Arial, sans-serif;
}

.application-container {
  text-align: left;
  background-color: #b1de87;
  min-height: 100%;
}

.container {
  width: 100%;
  padding-right: 8px;
  padding-left: 8px;
}

.title {
  display: block;
  font-weight: 600;
  width: 100%;
}

.subtitle {
  display: block;
  font-weight: 500;
  text-transform: none;
  width: 100%;
  font-size: 1.25rem;
  margin-top: 0px;
  margin-bottom: 0.5em;
}

.title-block {
  position: relative;
  max-width: 100%;
  margin-right: auto;
  margin-left: auto;
  background: none;
  text-align: center;
  padding-top: 30px;
  padding-bottom: 30px;
}

tr th.text-primary {
  color: #596f44 !important;
}

div .container-lg {
  margin-bottom: 4%;
}

main {
  padding-bottom: 30px;
}

.details {
  background-color: #ffffff;
}

input[type='checkbox'] {
  accent-color: var(--qa-green);
}
