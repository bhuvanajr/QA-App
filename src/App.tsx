import React from 'react';
import { ContainerFooter, ContainerHeader, QAOverview } from './components';

const App: React.FC = () => {
  return (
    <div className="application-container">
      <header>
        <ContainerHeader />
      </header>
      <main>
        <QAOverview />
      </main>
      <footer>
        <ContainerFooter />
      </footer>
    </div>
  );
};

export default App;
