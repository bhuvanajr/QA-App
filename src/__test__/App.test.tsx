import { render, screen } from '@testing-library/react';
import React from 'react';
import App from '../App';

test('should render correctly', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

test('should render page title', () => {
  render(<App />);
  const linkElement = screen.getByText(/Quotes Overview for you/i);
  expect(linkElement).toBeInTheDocument();
});
