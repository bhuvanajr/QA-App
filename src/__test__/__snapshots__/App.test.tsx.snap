// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="application-container"
  >
    <header>
      <div
        class="head-wrapper"
      >
        <div
          class="branding"
        >
          <a
            href="/"
          >
            <img
              alt="Brand logo"
              class="brand-image"
              src="/assets/qa_logo.png"
            />
          </a>
          QA Tool
        </div>
      </div>
    </header>
    <main
      class="container"
    >
      <div
        class="title-block"
      >
        <h1
          class="title"
        >
          The awesome QA Tool
        </h1>
        <h3
          class="subtitle"
        >
          Feel free to add new questions
        </h3>
      </div>
      <div
        class="mb-5 col-lg-10"
      >
        <div
          class="d-flex align-items-center mb-6"
        >
          <h2>
            Questions and Answers
          </h2>
          <button
            class="btn-icon"
            type="button"
          >
            <div
              class="icon ascending"
            />
          </button>
          <button
            class="btn-icon"
            type="button"
          >
            <div
              class="icon icon-delete"
            />
          </button>
        </div>
        <div
          class="accordion"
        >
          <div
            class="table-responsive rounded"
          >
            <table
              class="table mb-0"
            >
              <tbody>
                <tr
                  class="accordion accordion-item"
                >
                  <td
                    class="col-lg-1"
                  >
                    <a
                      href="#"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="icon icon-edit"
                      />
                    </a>
                  </td>
                  <th
                    class="text-primary text-break"
                    scope="row"
                  >
                    How to add a question
                  </th>
                  <td
                    class="col-lg-1"
                  >
                    <button
                      class="accordion-button collapsed"
                      data-bs-toggle="collapse"
                      type="button"
                    />
                  </td>
                </tr>
                <tr
                  class="details text-wrap"
                >
                  <td
                    class="p-0"
                    colspan="6"
                  >
                    <div
                      class="accordion-collapse collapse"
                    >
                      <div
                        class="px-3 py-3"
                      >
                        <div
                          class="row no-gutters"
                        >
                          <div
                            class="col-lg-11"
                          >
                            <dl
                              class="row no-gutters text-dark"
                            >
                              <dd
                                class="col-12 text-break"
                              >
                                Just use the form below
                              </dd>
                            </dl>
                          </div>
                        </div>
                      </div>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <form>
        <h2>
          Create a new question
        </h2>
        <div
          class="form-group row"
        >
          <label
            class="col-lg-8 col-form-label"
            for="question"
          >
            Question
          </label>
          <div
            class="col-lg-8"
          >
            <input
              autocomplete="off"
              class="form-control"
              name="question"
              placeholder="Enter your question here"
              rules="[object Object]"
            />
          </div>
        </div>
        <div
          class="form-group row"
        >
          <label
            class="col-lg-8 col-form-label"
            for="question"
          >
            Answer
          </label>
          <div
            class="col-lg-8"
          >
            <input
              autocomplete="off"
              class="form-control"
              name="answer"
              placeholder="Enter your answer here"
              rules="[object Object]"
            />
          </div>
        </div>
        <div
          class="col-auto my-2"
        >
          <div
            class="custom-control custom-checkbox mr-sm-2"
          >
            <input
              class="custom-control-input mx-2"
              name="delay"
              type="checkbox"
              value=""
            />
            <label
              class="custom-control-label"
              for="customControlAutosizing"
            >
              Introduce 5 second delay
            </label>
          </div>
        </div>
        <div
          class="col-auto my-1"
        >
          <button
            class="btn btn-primary btn-submit"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </main>
    <footer>
      <div
        class="container"
      >
        <div
          class="row "
        >
          <div
            class="col-sm-auto"
          >
            <ul
              class="nav flex-column flex-sm-row"
            >
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#"
                  rel="noreferrer"
                  target="_blank"
                >
                  Privacy
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#"
                  rel="noreferrer"
                  target="_blank"
                >
                  Cookies
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#"
                  rel="noreferrer"
                  target="_blank"
                >
                  <div
                    data-label="footer.copyright"
                  >
                    Â© QA Tool
                  </div>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
